multiplayer/src/services/simHost.ts

```diff
@@ -18,32 +18,25 @@ if (!pxt.react.getTilemapProject) {
     };
 }
 
-async function loadPackageAsync(runOpts: RunOptions, dependencies?: string[]) {
+async function loadPackageAsync(runOpts: RunOptions) {
     const verspec =
         runOpts.mpRole === "server" ? `pub:${runOpts.id}` : "empty:clientprj";
-    let host: pxt.Host;
-    let downloadPackagePromise: Promise<void>;
-    let installPromise: Promise<void>;
     const previousMainPackage = mainPkg();
-    if (previousMainPackage?._verspec == verspec) {
-        host = mainPkg().host();
-        downloadPackagePromise = Promise.resolve();
-        installPromise = Promise.resolve();
-    } else {
+    if (
+        previousMainPackage?._verspec !== verspec ||
+        verspec.startsWith("pub:S")
+    ) {
         const mp = mainPkg(true /** force refresh */);
+        // we want this to be cached only within the scope of a single call to loadpackageasync,
+        // as the file can be requested multiple times while loading.
+        mp.host().cacheStoreAsync(verspec, undefined!);
         mp._verspec = verspec;
-        downloadPackagePromise = mp
-            .host()
-            .downloadPackageAsync(mainPkg(), dependencies);
-        installPromise = mainPkg().installAllAsync();
+        await mp.host().downloadPackageAsync(mainPkg());
+        await mainPkg().installAllAsync();
```

Comment:
technically we were running these at the same time before, but it shouldn't matter as installAllAsync recursively grabs deps so it was already waiting for the script text network request anyways -- with this that will be available immediately  when downloadPackageAsync gets it